mdagent/tools/base_tools/preprocess_tools/uniprot.py

```diff
@@ -0,0 +1,1250 @@
+            return str(e)
+
+    async def _arun(self, query: str, primary_accession: str) -> str:
+        """use the tool asynchronously."""
+        raise NotImplementedError("This tool does not support asynchronous execution.")
+
+
+class GetActiveSites(BaseTool):
+    name = "GetActiveSites"
+    description = (
+        "Get the active sites known for a specific "
+        "protein, given the primary accession number. "
+        "Both the query string and primary accession "
+        "number are required. "
+    )
+    uniprot = QueryUniprot()
+
+    def _run(self, query: str, primary_accession: str = "") -> str:
+        """use the tool."""
+        try:
+            sites = self.uniprot.get_relevant_sites(query, primary_accession, "active")
+            return str(sites)
+        except Exception as e:
+            return str(e)
+
+    async def _arun(self, query: str, primary_accession: str) -> str:
+        """use the tool asynchronously."""
+        raise NotImplementedError("This tool does not support asynchronous execution.")
+
+
+class GetRelevantSites(BaseTool):
+    name = "GetRelevantSites"
+    description = (
+        "Get the relevant sites for a specific protein, "
+        "given the primary accession number. You must "
+        "provide the query string and primary accession "
+        "number. The relevant sites are sites that are "
+        "known to be important for the protein's function, "
+        "but are not necessarily active or binding sites."
+    )
+    uniprot = QueryUniprot()
+
+    def _run(self, query: str, primary_accession: str = "") -> str:
+        """use the tool."""
+        try:
+            sites = self.uniprot.get_relevant_sites(query, primary_accession, "sites")
+            return str(sites)
+        except Exception as e:
+            return str(e)
+
+    async def _arun(self, query: str, primary_accession: str) -> str:
+        """use the tool asynchronously."""
+        raise NotImplementedError("This tool does not support asynchronous execution.")
+
+
+class GetAllKnownSites(BaseTool):
+    name = "GetAllKnownSites"
+    description = (
+        "Get all known sites for a specific protein, "
+        "given the primary accession number. You must "
+        "provide the query string and primary accession "
+        "number. This tool is a one-stop shop to get all known sites "
+        "for the protein, including active sites, binding "
+        "sites, and other relevant sites."
+    )
+    uniprot = QueryUniprot()
+
+    def _run(self, query: str, primary_accession: str = "") -> str:
+        """use the tool."""
+        try:
+            active_sites = self.uniprot.get_relevant_sites(
+                query, primary_accession, "active"
+            )
+            binding_sites = self.uniprot.get_relevant_sites(
+                query, primary_accession, "binding"
+            )
+            sites = self.uniprot.get_relevant_sites(query, primary_accession, "sites")
+            return (
```

Comment:
If there are none in any of these three options can it be changed to something like: "No Binding Sites"?

Reply:
done